### PR TITLE
View Product Details Front End | Elliot Park

### PR DESCRIPTION
**Pull request by elliooo**
**Changes:**
- Created `ProductDetails` component that renders details on a clicked product
- Modified `App` component to conditionally render `ProductDetails` or `ProductList` based on state
- Minor code styling changes to `ProductListItem` to make the JSX more readable (i.e. breaking JSX elements with multiple props into multiple lines of code)

**Images and Gifs:**
view-details-front-end

### CODE REVIEW

File: client/components/app.jsx [hunk 1]
```diff
@@ -1,15 +1,40 @@
...
-      </>
+        {
+          this.state.view.name === 'catalog'
+            ? <ProductList setView={this.setView} />
+            : <ProductDetails setView={this.setView} viewParams={this.state.view.params} />
+        }
+      </div>
     );
+
+    // return <ProductDetails productId={6} />;
```
Comment (thebearingedge) [inline client/components/app.jsx]: Please do not turn in commented out code.

File: client/components/product-details.jsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      product: null,
+      isLoading: true
+    };
+  }
+
+  componentDidMount = () => {
+    // TODO: Replace with dynamic productId
```
Comment (thebearingedge) [inline client/components/product-details.jsx]: Please remove this

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: client/components/app.jsx (hunk 1 above)
Comment (elliooo) [reply client/components/app.jsx]: Resolved in 9e60879

File: client/components/product-details.jsx (hunk 2 above)
Comment (elliooo) [reply client/components/product-details.jsx]: Resolved in 9e60879

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**